lib/spack/spack/hooks/sbang.py

```diff
@@ -147,11 +151,47 @@ def install_sbang():
             spack.paths.sbang_script, sbang_path):
         return
 
-    # make $install_tree/bin and copy in a new version of sbang if needed
+    # make $install_tree/bin
     sbang_bin_dir = os.path.dirname(sbang_path)
     fs.mkdirp(sbang_bin_dir)
-    fs.install(spack.paths.sbang_script, sbang_path)
-    fs.set_install_permissions(sbang_bin_dir)
+
+    # get permissions for bin dir from configuration files
+    group_name = spack.package_prefs.get_package_group(spack.spec.Spec("all"))
+    config_mode = spack.package_prefs.get_package_dir_permissions(
+        spack.spec.Spec("all")
+    )
+
+    if group_name:
+        os.chmod(sbang_bin_dir, 0o775)
+    else:
+        os.chmod(sbang_bin_dir, 0o755)
```

Comment:
I was thinking more like:

```suggestion
    if group_name:
        os.chmod(sbang_bin_dir, config_mode)   # Use package directory permissions
    else:
        fs.set_install_permissions(sbang_bin_dir)
```